compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaServiceDeclarationSymbol.java

```diff
@@ -216,4 +218,42 @@ public BallerinaServiceDeclarationSymbol build() {
                                                          this.qualifiers, this.annots, this.bSymbol, this.context);
         }
     }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(this.getInternalSymbol().getName().getValue(),
+                this.getLocation().map(Location::lineRange).orElse(null));
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (!(obj instanceof ServiceDeclarationSymbol)) {
+            return false;
+        }
+
+        BallerinaServiceDeclarationSymbol symbol = (BallerinaServiceDeclarationSymbol) obj;
+        return nameEquals(symbol.getInternalSymbol().getName().getValue())
+                && isSameLocation(this.getLocation(), symbol.getLocation());
+    }
+
+    @Override
+    public boolean nameEquals(String name) {
+        String symbolName = this.getInternalSymbol().getName().getValue();
+        if (name.equals(symbolName)) {
+            return true;
+        }
+        return unescapedUnicode(name).equals(unescapedUnicode(symbolName));
+    }
```

Comment:
This method is for comparing the actual name of a symbol. Since we don't have a public name for service symbols, let's not override this.